pkg/datapath/cells.go

```diff
@@ -92,7 +92,16 @@ func newDatapath(params datapathParams) types.Datapath {
 			return nil
 		}})
 
-	return linuxdatapath.NewDatapath(datapathConfig, iptablesManager, params.WgAgent)
+	datapath := linuxdatapath.NewDatapath(datapathConfig, iptablesManager, params.WgAgent)
+
+	params.LC.Append(hive.Hook{
+		OnStart: func(hive.HookContext) error {
+			datapath.Node().RestoreNodeIDs()
```

Comment:
note: `RestoreNodeIDs` will enforce opening the BPF map if it's not already opened - which is the case with this move because the OnStart lifecycle hook will be executed before the initialization of the BPF map in the daemon init.

see https://github.com/cilium/cilium/blob/main/pkg/ebpf/map.go#L177-L181

anyway - i'll come up with an additional PR which will refactor the node map to be provided as hive cell after this one got merged. This will make this dependency visible.